plugins/env/env.go

```diff
@@ -44,6 +61,10 @@ func (plugin) NewTaskPlugin(options plugins.TaskPluginOptions) (plugins.TaskPlug
 		return nil, engines.ErrContractViolation
 	}
 
+	for k, v := range pl.extraVars {
+		p.Env[k] = v
+	}
```

Comment:
you should set `TASKID` and `RUNID` based on `options.Context`, I think it's in there somewhere in the `TaskContext`